src/homeWork17/home_work17.ts

```diff
@@ -0,0 +1,348 @@
+
+//"Поточні відвідувачі":
+
+//Зберігає інформацію про відвідувачів, включаючи їхні імена та контактні дані.
+
+//Можливість оповіщення відвідувачів за 15 хвилин до закриття і перед відходом.
+
+//"Клієнти":
+
+//Дані клієнтів зберігаються у Відділу реклами.
+
+//Відділ реклами використовує цей список для розсилки новин про зоопарк і рекламні акції.
+
+//"Відділ реклами":
+
+//Відповідає за маркетингові та рекламні заходи.
+
+//Використовує список клієнтів для розсилки новин про зоопарк і рекламні акції.
+
+//"Виручка":
+
+//Каса збирає дані про виручку за день.
+
+//Ці дані передаються в Бухгалтерію.
+
+//"Бухгалтерія":
+
+//Відповідає за фінансове управління зоопарку.
+
+//Розпоряджається бюджетом, включно з оплатою співробітників, закупівлею корму для тварин і обслуговуванням зоопарку.
+
+//Зберігає дані про всіх співробітників, тварин і виплати.
+
+//Можливість генерувати фінансові звіти.
+
+//"Адміністрація":
+
+//Відповідає за управління співробітниками і тваринами.
+
+//Може додавати і видаляти співробітників і тварин.
+
+//Створює сповіщення про рекламні акції та інші важливі події в зоопарку.
+
+//"Тварини":
+
+//Включає в себе інформацію про кожну тварину, таку як вид, ім'я, вік, здоров'я та інші характеристики.
+
+//"Співробітники":
+
+//Адміністрація може додавати і видаляти співробітників.
+
+//Співробітники можуть мати різні посади та обов'язки, які слід враховувати.
+
+//"Бюджет":
+
+//Бухгалтерія розпоряджається бюджетом і стежить за фінансами зоопарку.
+
+//Можливість вести бюджетний облік і надавати фінансові звіти.
+interface IEmployees{
+    name:string;
+    age:number;
+    position:string;
+
+}
+
+interface IAnimals{
+    kind:string;
+    name:string;
+    age:number;
+    health:string;
+}
+
+interface IVisitors{
+    name:string;
+    contactData:string;
+}
+
+interface ICustomer extends IVisitors{}
+
+
+interface ITickets{
+    type:'adult' | 'child' | 'family';
+    value:number;
+}
+
+// interface IBudget{
+//     budgetAll:number,
+   
+// }
+
+
+// interface Receipt{
+
+// }
+class CashRegister {
+    private visitor:IVisitors[] = [];
+    private customer:ICustomer[] = [];
+    protected amountTicket:number = 0;
+  
+    constructor(private closingTime: Date) {
+       
+    }
+
+    sellTickets(tickets:{type:string,value:number}):void{
+     
+      if (tickets.type = 'adult'){
+        this.amountTicket += tickets.value;
+        
+      }else if(tickets.type = 'child'){
+        this.amountTicket += tickets.value;
+       
+        
+      }else if(tickets.type = 'family'){
+        this.amountTicket += tickets.value;
+      
+      }else{
+        console.log('No tickets sold');
+      }
+      console.log(`Tickets sold for the amount $${this.amountTicket}`);
+  
+ 
+    }
+    addPeople(visitor:IVisitors):void{
+         
+        this.visitor.push(visitor);
+        this.customer.push({...visitor});
+        console.log(visitor);
+       
+    }
+  
+    notificationBeforeClosing():void{
```

Comment:
The code needs to be refactored, no doubt about it. You can seize the opportunity to sharpen your expertise with the Observer pattern.